servers/services/src/main/java/org/projectnessie/services/rest/BaseResource.java

```diff
@@ -84,7 +84,7 @@ protected void doOps(String branch, String hash, String message, List<org.projec
     } catch (IllegalArgumentException e) {
       throw new NessieNotFoundException("Invalid hash provided.", e);
     } catch (ReferenceConflictException e) {
-      throw new NessieConflictException("Failed to commit data. Provided hash does not match current value.", e);
+      throw new NessieConflictException("Failed to commit data. " + e.getMessage(), e);
```

Comment:
Can we have a richer error? We've gone from 'guessing and wrong in some cases' to hoping that the underlying exception has a useful value. Eg 'Usually this means the provided hash does not match current value, {e.getMessage()} may provide more context'